payroll-processor-functions/test/payroll-processor-functions-test/Features/Employees/EmployeesTriggerTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json;
+using NSubstitute;
+using PayrollProcessor.Functions.Features.Employees;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace PayrollProcessor.Functions.Tests.Features.Employees
+{
+    public class EmployeesTriggerTests
+    {
+        private HttpRequest DefaultHttpRequest { get; }
+        private CloudTable Table { get; }
+        private ILogger Logger { get; }
+        private EmployeesTrigger Trigger { get; }
+
+        public EmployeesTriggerTests()
+        {
+            DefaultHttpRequest = Substitute.For<HttpRequest>();
+
+            Logger = Substitute.For<ILogger>();
+
+            Table = TableStorageFixtures.CreateTable("employees");
+
+            Trigger = new EmployeesTrigger();
+        }
+
+        [Fact]
+        public async void GetEmployees_Should_Return_All_Employees()
+        {
+            var entity = new EmployeeEntity { PartitionKey = "employees", RowKey = Guid.NewGuid().ToString("n") };
+
+            var segment = TableStorageFixtures.CreateSegment(new[] { entity });
+
+            Table
+                .ExecuteQuerySegmentedAsync(Arg.Any<TableQuery<EmployeeEntity>>(), Arg.Any<TableContinuationToken>())
+                .Returns(segment);
+
+            var result = await Trigger.GetEmployees(DefaultHttpRequest, Table, Logger);
+
+            var response = result as OkObjectResult;
+
+            string resultValue = response.Value.ToString();
+
+            var employeesResult = JsonConvert.DeserializeObject<IEnumerable<Employee>>(resultValue);
+
+            employeesResult.Should().HaveCount(1);
+            employeesResult.First().Id.Should().Be(Guid.Parse(entity.RowKey));
+        }
+    }
+
+    public static class TableStorageFixtures
+    {
+        public static CloudTable CreateTable(string tableName) =>
+            Substitute.For<CloudTable>(new Uri($"http://127.0.0.1:10002/devstoreaccount1/{tableName}"));
+
+        public static TableQuerySegment<T> CreateSegment<T>(IEnumerable<T> data)
+        {
+            // https://github.com/Azure/azure-storage-net/issues/619#issuecomment-364090291
+
+            var ctor = typeof(TableQuerySegment<T>)
```

Comment:
This type doesn't have a public constructor, which makes it very hard to create instances of :)

This approach is something another dev came up with and it works well.

Reply:
Very creative use of reflection!